### PR TITLE
Api

### LINKED ISSUE #8: Extract data from National Rail
### Task

- [x] extract data from national rail
- [x] create README.md file explaining the folder

### User Story

As a developer, I want to be able to extract the data from National Rail so I can read and transform it

### LINKED ISSUE #9: Extract data from Realtime Trains
### Task

- [x] Create an extract file for Realtime Trains
- [x] Create README.md file for Realtime Train

### User Story

As a developer, I want to extract the Realtime Trains data so I can read and transform the data

### PR DESCRIPTION
**Pull request by A-Faris**
Added initial extract file, README and test file for national rail and realtime trains apis.

closes #9, closes #8

### CODE REVIEW

File: national_rail/extract.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+def get_data_from_api() -> None:
+    """Gets the data from the national rail api and saves it to a file"""
+    load_dotenv()
+
+    url = "https://api1.raildata.org.uk/1010-knowlegebase-incidents-xml-feed1_0/incidents.xml"
+
+    headers = {
+        "x-apikey": ENV["APIKEY"],
+        "User-Agent": ""
```
Comment (fm1psy) [inline national_rail/extract.py:17]: Why is this User agent empty? Is it needed?

File: national_rail/extract.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+from os import environ as ENV
+
+from requests import get
+from dotenv import load_dotenv
+
+
+def get_data_from_api() -> None:
+    """Gets the data from the national rail api and saves it to a file"""
+    load_dotenv()
```
Comment (fm1psy) [inline national_rail/extract.py:11]: now why is this here? I think it should be in the if name block

File: national_rail/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Configuration
+
+Before running the script, you need to set up your National Rail API credentials. Create a new file called `.env` in the `national_rail` directory and add the following lines, replacing `YOUR_API_KEY` with your actual API key:
+
+```text
+NATIONAL_RAIL_API_KEY=YOUR_API_KEY
+```
+
+The script will automatically load the API key from this file.
```
Comment (FarihaChoudhury) [inline national_rail/README.md:44]: readme looks good. I like that you added installation guide

File: national_rail/extract.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    url = "https://api1.raildata.org.uk/1010-knowlegebase-incidents-xml-feed1_0/incidents.xml"
+
+    headers = {
+        "x-apikey": ENV["APIKEY"],
+        "User-Agent": ""
+    }
+
+    response = get(url, headers=headers, timeout=10)
+
+    with open("data.xml", "w", encoding="utf-8") as file:
```
Comment (fm1psy) [inline national_rail/extract.py:22]: I like the use of with here, very modern.

File: realtime_trains/extract.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+from requests.auth import HTTPBasicAuth
+from requests import get
+from dotenv import load_dotenv
+
+CRS = "LST"
+
+
+def get_data_from_api() -> None:
+    """Gets the data from the realtime trains api and saves it to a file"""
+    load_dotenv()
```
Comment (fm1psy) [inline realtime_trains/extract.py:16]: I feel as though this load_dotenv belongs in the if name block.

File: national_rail/extract.py (hunk 1 above)
Comment (A-Faris) [reply national_rail/extract.py:17]: Yes so that the request can work. It is probably used as a tag

**Review by fm1psy — COMMENTED** [review]
Overall good work, just a couple of questions for you to answer thank you.

File: realtime_trains/extract.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    url = path.join(base_url, CRS, yesterday)
+
+    username = ENV["USERNAME"]
+    password = ENV["PASSWORD"]
+
+    response = get(url, auth=HTTPBasicAuth(username, password), timeout=10)
+
+    with open("data.json", "w", encoding="utf-8") as file:
+        json.dump(response.json(), file, indent=4)
+
```
Comment (FarihaChoudhury) [inline realtime_trains/extract.py:30]: good stuff in this function.

**Review by FarihaChoudhury — COMMENTED** [review]
Good work

File: README.md [hunk 7]
```diff
@@ -4,3 +4,8 @@
 
 # Railway-Tracker
 
+https://www.realtimetrains.co.uk/about/developer/pull/docs/
+
+```
+api.rtt.io/api/v1/json/search/<filter>
```
Comment (FarihaChoudhury) [inline README.md]: I don't think we need this here

File: national_rail/extract.py (hunk 2 above)
Comment (A-Faris) [reply national_rail/extract.py:11]: Yes that's possible but if this function is called form another file, they wouldn't need to put in load_dotenv()

File: national_rail/README.md (hunk 3 above)
Comment (A-Faris) [reply national_rail/README.md:44]: Thank you, I appreciate that

File: national_rail/extract.py (hunk 4 above)
Comment (A-Faris) [reply national_rail/extract.py:22]: Thank you

File: realtime_trains/extract.py (hunk 5 above)
Comment (A-Faris) [reply realtime_trains/extract.py:16]: Same reason as I gave above

File: realtime_trains/extract.py (hunk 6 above)
Comment (A-Faris) [reply realtime_trains/extract.py:30]: Thank you

File: README.md (hunk 7 above)
Comment (A-Faris) [reply README.md]: Fixed

**Comment by FarihaChoudhury** [comment]
Good work on accessing data for national rail and real time trains apis

**Review by FarihaChoudhury — APPROVED** [review]
